src/network/serverpackethandler.cpp

```diff
@@ -1358,7 +1358,8 @@ void Server::handleCommand_Interact(NetworkPacket* pkt)
 		Check that target is reasonably close
 		(only when digging or placing things)
 	*/
-	if (action == 0 || action == 2 || action == 3) {
+	if ((!isSingleplayer() && !g_settings->getBool("disable_anticheat")) &&
+			(action == 0 || action == 2 || action == 3)) {
```

Comment:
@est31 so you are claiming the server can change from single-player to multiplayer without needing to shutdown the server* while running?
____
in any case, I realize that it would be hard if not impossible to update the cache** when changing from single-player to multiplayer mode if it is place in here, it would need to be in a place where it would only run once when starting up a server (whether it's in single-player or multiplayer mode like in server.cpp. however it is not only bad, it would be stupid to put the cache in server.cpp just for this)

I guess there's no way to simply and cleanly cache that statement.

* exit to menu and exit to os both shutdown the server
** the output of `(!isSingleplayer() && !g_settings->getBool("disable_anticheat"))`